File: NewsfeedClient/Models/Digest.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace NewsfeedClient.Models
{
    public class Digest
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public Boolean IsPublic { get; set; }

        public int CreatorId { get; set; }
        public User Creator { get; set; }

        public List<Subscription> Subscribers { get; set; }
        public List<DigestSource> DigestSources { get; set; }

        public Digest()
        {
            Subscribers = new List<Subscription>();
            DigestSources = new List<DigestSource>();
        }
    }
}

File: NewsfeedClient/ClientApp/src/app/shared/models/digest.ts
import { Injectable } from '@angular/core';
import { User } from './user';
import { Source } from './source';

@Injectable()
export class Digest {

  constructor(
    public name: string,
    public creator: string,
    public isPublic: boolean,
    public id: number,
    public sources: Source[]) {
  }
}
